### ISSUE TITLE
hyperkit: IP address never found in dhcp leases file Temporary Error: Could not find an IP address for <mac>

### ISSUE THREAD

**Issue by colangelo** [kubernetes/minikube#4206]
Hi,
minikube does not start on my mac, I tried multiple times to remove and reinstall Docker Desktop and docker-machine-driver-hyperkit without solving the problem.
It's not related to version 1.0.1, because the problem was present also with version 1.0.0 and maybe also with previous versions that I haven't tested.

When I lanch minikube the **/usr/local/bin/docker-machine-driver-hyperkit** process appears in the process list, does not use CPU, and it's still present for some time after the command exited.

Can you help me?

**The exact command to reproduce the issue**:
```
minikube stop; minikube delete ; rm -rf ~/.minikube
minikube config set vm-driver hyperkit
minikube start -v 10
```

**The full output of the command that failed**:
```
😄  minikube v1.0.1 on darwin (amd64)
💿  Downloading Minikube ISO ...
 142.88 MB / 142.88 MB [============================================] 100.00% 0s
🤹  Downloading Kubernetes v1.14.1 images in the background ...
🔥  Creating hyperkit VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...
Found binary path at /usr/local/bin/docker-machine-driver-hyperkit
Launching plugin server for driver hyperkit
Plugin server listening at address 127.0.0.1:54243
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minikube) Calling .GetMachineName
(minikube) Calling .DriverName
Creating CA: /Users/user/.minikube/certs/ca.pem
Creating client certificate: /Users/user/.minikube/certs/cert.pem
Running pre-create checks...
(minikube) Calling .PreCreateCheck
(minikube) Calling .GetConfigRaw
Creating machine...
(minikube) Calling .Create
(minikube) Downloading /Users/user/.minikube/cache/boot2docker.iso from file:///Users/user/.minikube/cache/iso/minikube-v1.0.1.iso...
(minikube) DBG | 2019/05/04 17:33:26 [INFO] Creating ssh key...
(minikube) DBG | 2019/05/04 17:33:26 [INFO] Creating raw disk image...
(minikube) DBG | Writing magic tar header
(minikube) DBG | Writing SSH key tar header
(minikube) DBG | Mounting boot2docker.iso
(minikube) DBG | executing: &{/usr/bin/hdiutil [hdiutil attach /Users/user/.minikube/machines/minikube/boot2docker.iso -mountpoint /Users/user/.minikube/machines/minikube/b2d-image] []  <nil> 0xc42000e018 0xc42000e02
0 [] <nil> <nil> <nil> <nil> <nil> false [] [] [] [] <nil> <nil>} attach /Users/user/.minikube/machines/minikube/boot2docker.iso -mountpoint /Users/user/.minikube/machines/minikube/b2d-image
(minikube) /dev/disk3                                           /Users/user/.minikube/machines/minikube/b2d-image
(minikube) DBG | Extracting Kernel Options...
(minikube) DBG | Extracted Options "loglevel=3 user=docker console=ttyS0 console=tty0 noembed nomodeset norestore waitusb=10 systemd.legacy_systemd_cgroup_controller=yes base host=minikube"
(minikube) DBG | Extracting /Users/user/.minikube/machines/minikube/b2d-image/boot/bzImage into /Users/user/.minikube/machines/minikube/bzImage
(minikube) DBG | Extracting /Users/user/.minikube/machines/minikube/b2d-image/boot/initrd into /Users/user/.minikube/machines/minikube/initrd
(minikube) DBG | Unmounting boot2docker.iso
(minikube) DBG | executing: &{/usr/bin/hdiutil [hdiutil detach /Users/user/.minikube/machines/minikube/b2d-image] []  <nil> 0xc42000e018 0xc42000e020 [] <nil> <nil> <nil> <nil> <nil> false [] [] [] [] <nil> <nil>} de
tach /Users/user/.minikube/machines/minikube/b2d-image
(minikube) "disk3" ejected.
(minikube) Using UUID f59fbfd4-6e81-11e9-bb44-f01898eab608
(minikube) Generated MAC 4e:e8:f0:ac:b:3f
(minikube) Starting with cmdline: loglevel=3 user=docker console=ttyS0 console=tty0 noembed nomodeset norestore waitusb=10 systemd.legacy_systemd_cgroup_controller=yes base host=minikube

💣  Unable to start VM: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f
Temporary Error: Could not find an IP address for 4e:e8:f0:ac:b:3f

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new
```

**The output of the `minikube logs` command**:
```
💣  command runner: getting ssh client for bootstrapper: Error dialing tcp via ssh client: dial tcp :22: connect: connection refused

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new

```

**The operating system version**:

macOS Mojave 10.14.4 (18E226)
Darwin mac.local 18.5.0 Darwin Kernel Version 18.5.0: Mon Mar 11 20:40:32 PDT 2019; root:xnu-4903.251.3~3/RELEASE_X86_64 x86_

**Comment by tstromberg** [comment]
This is a duplicate of #1926 - the solution is to upgrade your hyperkit driver:

`curl -LO https://storage.googleapis.com/minikube/releases/latest/docker-machine-driver-hyperkit && sudo install -o root -g wheel -m 4755 docker-machine-driver-hyperkit /usr/local/bin/`

Then run `minikube delete && minikube start --vm-driver=hyperkit` to recreate your VM.

If this is still broken, please re-open the bug, and include the output of:

`ls -la $(which docker-machine-driver-hyperkit)`

I will make sure we provide a more user friendly error message for this case. Thanks!

**Comment by colangelo** [comment]
I've tryed what you suggest many times, but I'm still having the exact same issue.

$ ls -la $(which docker-machine-driver-hyperkit)
-rwsr-xr-x  1 root  wheel  27785684 May 15 19:17 /usr/local/bin/docker-machine-driver-hyperkit*

**Comment by tstromberg** [comment]
Thanks for the update! Do you mind providing updated output for `minikube start --alsologtostderr -v=8 ` as well as the contents of `/var/db/dhcpd_leases`? 

The newer version of the hyperkit driver (which you appear to be on) should output a bit more information on what DHCP entries it sees in that file.

**Comment by tstromberg** [comment]
Regrettably, there isn't enough information in this issue to make it actionable, and a long enough duration has passed, so this issue is likely difficult to replicate. If you can provide any additional details, such as:

- The exact command line used
- The full output of the command
- The full output of "minikube logs"

Please feel free to do so at any point. Thank you for sharing your experience!

**Comment by snoopdouglas** [comment]
This might be a naive assumption, but on Mojave there is no `/var/db/dhcp_leases`. There is, however, `/var/db/dhcpclient/leases`.

**Comment by tstromberg** [comment]
Can you share the output of:

`docker-machine-driver-hyperkit version`

and:

`minikube start -v=7`

**Comment by amctavish** [comment]
Also experiancing this issue.

OS: macOS Mojave 10.14.6

`hyperkit: v0.20190201-11-gc0dd46`

`docker-machine version 0.16.1, build cce350d7`

```
> minikube start --vm-driver=hyperkit
😄  minikube v1.3.0 on Darwin 10.14.6
💿  Downloading VM boot image ...
    minikube-v1.3.0.iso.sha256: 65 B / 65 B [=====================] 100.00% 0s
    minikube-v1.3.0.iso: 131.07 MiB / 131.07 MiB [===============] 100.00% 45s
🔥  Creating hyperkit VM (CPUs=2, Memory=2000MB, Disk=20000MB) ...
E0808 20:36:34.330343   97870 start.go:723] StartHost: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory

💣  Unable to start VM: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary Error: open /var/db/dhcpd_leases: no such file or directory

Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

```
> docker-machine-driver-hyperkit version
This is a Docker Machine plugin binary.
Plugin binaries are not intended to be invoked directly.
Please use this plugin through the main 'docker-machine' binary.
(API version: 1)
```

`minikube start -v=7`: didn't work as I don't have VirtualBox installed (really don't want to have to install VirtualBox)
```
💣  Unable to start VM
❌  Error: [VBOX_NOT_FOUND] create: precreate: VBoxManage not found. Make sure VirtualBox is installed and VBoxManage is in the path
```

As @snoopdouglas mentioned using Mojave there is `/var/db/dhcpclient/leases` but no `/var/db/dhcpd_leases`

**Comment by rkrdo** [comment]
I'm also having the same issue using Mojave,

The output of `minikube start --vm-driver hyperkit -v=7` is:

```
😄  minikube v1.3.0 on Darwin 10.14.5
💿  Downloading VM boot image ...
    minikube-v1.3.0.iso.sha256: 65 B / 65 B [=====================] 100.00% 0s
    minikube-v1.3.0.iso: 131.07 MiB / 131.07 MiB [================] 100.00% 1s
🔥  Creating hyperkit VM (CPUs=2, Memory=2000MB, Disk=20000MB) ...
Found binary path at /usr/local/bin/docker-machine-driver-hyperkit
Launching plugin server for driver hyperkit
Plugin server listening at address 127.0.0.1:59942
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(minikube) Calling .GetMachineName
(minikube) Calling .DriverName
Creating CA: /Users/ricardo/.minikube/certs/ca.pem
Creating client certificate: /Users/ricardo/.minikube/certs/cert.pem
Running pre-create checks...
(minikube) Calling .PreCreateCheck
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) Calling .GetConfigRaw
Creating machine...
(minikube) Calling .Create
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | ERROR: logging before flag.Parse: I0808 18:31:48.583791    8388 drivers.go:96] Making disk image using store path: /Users/ricardo/.minikube
(minikube) Downloading /Users/ricardo/.minikube/cache/boot2docker.iso from file:///Users/ricardo/.minikube/cache/iso/minikube-v1.3.0.iso...
(minikube) DBG | ERROR: logging before flag.Parse: I0808 18:31:48.708304    8388 drivers.go:103] Creating ssh key: /Users/ricardo/.minikube/machines/minikube/id_rsa...
(minikube) DBG | ERROR: logging before flag.Parse: I0808 18:31:48.783841    8388 drivers.go:109] Creating raw disk image: /Users/ricardo/.minikube/machines/minikube/minikube.rawdisk...
(minikube) DBG | Writing magic tar header
(minikube) DBG | Writing SSH key tar header
(minikube) DBG | ERROR: logging before flag.Parse: I0808 18:31:48.784123    8388 drivers.go:123] Fixing permissions on /Users/ricardo/.minikube/machines/minikube ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | clean start, hyperkit pid file doesn't exist: /Users/ricardo/.minikube/machines/minikube/hyperkit.pid
(minikube) DBG | Using UUID b0df7f16-ba34-11e9-8891-8c85907a2ecb
(minikube) DBG | Generated MAC e:3a:ab:fb:c9:6b
(minikube) DBG | Starting with cmdline: loglevel=3 user=docker console=ttyS0 console=tty0 noembed nomodeset norestore waitusb=10 systemd.legacy_systemd_cgroup_controller=yes base host=minikube
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | 2019/08/08 18:31:49 fcntl(F_PUNCHHOLE) failed: host filesystem does not support sparse files: Operation not permitted
(minikube) DBG | 2019/08/08 18:31:49 rdmsr to register 0x140 on vcpu 0
(minikube) DBG | 2019/08/08 18:31:49 rdmsr to register 0x140 on vcpu 1
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | 2019/08/08 18:32:06 rdmsr to register 0x64e on vcpu 0
(minikube) DBG | 2019/08/08 18:32:06 rdmsr to register 0x34 on vcpu 0
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
(minikube) DBG | exe=/usr/local/bin/docker-machine-driver-hyperkit uid=0
(minikube) DBG | hyperkit pid from json: 8416
(minikube) DBG | Searching for e:3a:ab:fb:c9:6b in /var/db/dhcpd_leases ...
E0808 18:32:51.347041    8366 start.go:723] StartHost: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory

💣  Unable to start VM: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
Temporary Error: open /var/db/dhcpd_leases: no such file or directory

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

And the output of `docker-machine-driver-hyperkit version` is:

```
version: v1.3.0
commit: 43969594266d77b555a207b0f3e9b3fa1dc92b1f
```

`minikube logs` returns:

```
💣  command runner: getting ssh client for bootstrapper: Error dialing tcp via ssh client: dial tcp :22: connect: connection refused

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

and like other folks in this thread, I don't have a `/var/db/dhcp_leases` file

**Comment by colangelo** [comment]
@rkrdo Are you using a vpn at the same time, or have a vpn client installed?
In my case the problem is probably related to some weird interaction between minikube and vpn software. 
But for me it's tricky to determine because it's inconsistent, or at least i didn't recognize a partern yet.

**Comment by tyagi** [comment]
Same with me too. Missing ```/var/db/dhcpd_leases``` file on my mac.

#### Details 
* OS: High Sierra [10.13.6]
* Same Exception Trace
* HyperKit - latest 
```
➜  ~ docker-machine-driver-hyperkit version
version: v1.3.0
commit: 43969594266d77b555a207b0f3e9b3fa1dc92b1f
```
* Minikube - latest
```
➜  ~ minikube version
minikube version: v1.3.0
commit: 43969594266d77b555a207b0f3e9b3fa1dc92b1f
```

Note: 
* I can start Minikube using Virtualbox.
* I do have Cisco AnyConnect software installed for connecting to VPN but I am not using it when starting Minikube.
* Laptop is company managed as in few policies are in place like Siri being unavailable.

Please let me know if you need any other information to fix this issue, happy to help.

**Comment by rkrdo** [comment]
> @rkrdo Are you using a vpn at the same time, or have a vpn client installed?
> In my case the problem is probably related to some weird interaction between minikube and vpn software.
> But for me it's tricky to determine because it's inconsistent, or at least i didn't recognize a partern yet.

@colangelo thank you for the suggestion! it turned out it to be [Little Snitch](https://www.obdev.at/products/littlesnitch/index.html) that was causing the issue somehow.

**Comment by colangelo** [comment]
Thanks @rkrdo, good to know!

I also have Little Snitch installed, and most of the times i have no issues, but sometimes especially a few months ago, if i'd recently connected or disconnetted from the vpn i was getting that error.

My opinion is that this issue is probably related to one ore more third party softwares that interact with the network stack, and interact badly with minikube (or hyperkit).

**Comment by tstromberg** [comment]
Update: This issue is still possible, but minikube v1.4 now retries the creation process, so you are much less likely to see the problem in the wild. That said, there are still programs like "Little Snitch" that may interfere, so I am leaving this open.

**Comment by tstromberg** [comment]
Also adding flake tag because this issue is the source of test flakes:

https://storage.googleapis.com/minikube-builds/logs/4980/HyperKit_macOS.txt

**Comment by cequencer** [comment]
I am also have this identical issue just to reflect that I am using the latest and greatest.  I did not where to file my issue in the beginning, so I landed my issue here:
https://github.com/machine-drivers/docker-machine-driver-hyperkit/issues/15

**Comment by tstromberg** [comment]
Now that we have great output for our test failures, I have a theory: with a modern hyperkit driver: once you've setup 253 VM's, you will hit this failure condition. Basically:

- The DHCP server (bootpd) is allocated to only provide 253 IP's
- The hyperkit driver generates a unique MAC address based on the VM's unique identifier
- bootpd preserves the last lease for every MAC address seen

My suggested workaround:

Once the lease count goes over a certain limit (>240), the driver could check if bootpd is running, and if not, flush expired leases.

It's worth noting that bootpd is only typically active for just long enough to start a hyperkit VM.

**Comment by tstromberg** [comment]
Example: https://storage.googleapis.com/minikube-builds/logs/5584/HyperKit_macOS.txt

**Comment by sshambar** [comment]
I encountered this error, but after some digging I discovered in my case the cause was bootpd being disabled (I had installed the ISC dhcpd at some point).  Running the following resolved the issue:

> sudo launchctl enable system/com.apple.bootpd

**Comment by davix** [comment]
Hi I'm running the same issue and I don't have "Little Snitch" or bootpd installed.
Any workaround?

```
➜  ~ minikube start -v=7
😄  minikube v1.5.2 on Darwin 10.14.6
✨  Automatically selected the 'hyperkit' driver
🔥  Creating hyperkit VM (CPUs=2, Memory=2000MB, Disk=20000MB) ...
🔄  Retriable failure: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary error: open /var/db/dhcpd_leases: no such file or directory
🔥  Deleting "minikube" in hyperkit ...
...

💣  Unable to start VM: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary error: open /var/db/dhcpd_leases: no such file or directory

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

**Comment by giuseppefg83** [comment]
my /var/db/dhcpd_leases contains :

{
	name=minikube
	ip_address=192.168.64.14
	hw_address=1,56:f:92:91:49:cf
	identifier=1,56:f:92:91:49:cf
	lease=0x5ded0faf
}
{
	name=minikube
	ip_address=192.168.64.13
	hw_address=1,ea:e4:76:e6:38:8f
	identifier=1,ea:e4:76:e6:38:8f
	lease=0x5ded0df1
}
{
	name=minikube
	ip_address=192.168.64.12
	hw_address=1,7a:8e:f2:a0:c2:f3
	identifier=1,7a:8e:f2:a0:c2:f3
	lease=0x5ded0dad
}
{
	name=minikube
	ip_address=192.168.64.11
	hw_address=1,4e:14:6:3d:86:cb
	identifier=1,4e:14:6:3d:86:cb
	lease=0x5ded0c85
}
{
	name=minikube
	ip_address=192.168.64.10
	hw_address=1,76:6f:a7:e2:e4:f2
	identifier=1,76:6f:a7:e2:e4:f2
	lease=0x5ded0c3c
}
{
	name=minikube
	ip_address=192.168.64.9
	hw_address=1,52:8:75:9b:91:a3
	identifier=1,52:8:75:9b:91:a3
	lease=0x5decef57
}
{
	name=minikube
	ip_address=192.168.64.8
	hw_address=1,52:2f:b5:e1:da:b6
	identifier=1,52:2f:b5:e1:da:b6
	lease=0x5decee09
}
{
	name=minikube
	ip_address=192.168.64.7
	hw_address=1,de:6d:92:a5:bf:2f
	identifier=1,de:6d:92:a5:bf:2f
	lease=0x5deced57
}
{
	name=minikube
	ip_address=192.168.64.6
	hw_address=1,a6:d3:4c:1b:8e:11
	identifier=1,a6:d3:4c:1b:8e:11
	lease=0x5deced01
}
{
	name=minikube
	ip_address=192.168.64.5
	hw_address=1,36:df:ef:74:46:7
	identifier=1,36:df:ef:74:46:7
	lease=0x5dececc7
}
{
	name=minikube
	ip_address=192.168.64.4
	hw_address=1,32:3d:f5:ad:10:c0
	identifier=1,32:3d:f5:ad:10:c0
	lease=0x5dece981
}
{
	name=minikube
	ip_address=192.168.64.3
	hw_address=1,6:e6:d8:2e:66:a7
	identifier=1,6:e6:d8:2e:66:a7
	lease=0x5dece937
}
{
	name=minikube
	ip_address=192.168.64.2
	hw_address=1,7e:77:3b:7b:46:7d
	identifier=1,7e:77:3b:7b:46:7d
	lease=0x5dece8f8
}



I have solved by clear this configuration and recreate minikube:
minikube delete
minikube start --vm-driver=hyperkit

**Comment by tstromberg** [comment]
We still see this from time to time in our integration tests. As best as I can tell, it's reflective of a class of bugs with VM startup in hyperkit, but all attempts to reliably replicate it have failed.

**Comment by jaydarius** [comment]
minikube: v1.6.2
Darwin: 10.14.6
hyperkit: v0.20190201-11-gc0dd46

I experienced this same issue.

Resolution was enabling my company VPN through "Pulse Secure" app.

**Comment by nickwilmes** [comment]
Seeing same issue with a fresh install of docker for mac and minikube.  It seems to be consistent (I tried uninstalling and reinstalling at least 5 times and got the same error each time).  Also using mohave and do have a company VPN software installed if that is the cause (f5 Big-IP edge client).

**Comment by zarenner** [comment]
We're hitting this on the [Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops) macOS images, either 10.14 or 10.15. 

One curiosity is that multiple times I've tested this, the first couple runs work fine, but then this error reproduces 100% of the time after that. That seems like it sort of meshes with @tstromberg's theory that a lease limit is being hit... but AFAIK those hosted VMs should be completely isolated so that's really weird. Could just be a coincidence.

**Comment by akram** [comment]
I faced the issue also, and solved it by:
- disabling internet sharing in OSX (I don't know if it helped solving)
- editing the /var/db/dhcpd_leases and changed the entry named "minishift" by setting the fields `hw_address=1,32:76:8c:29:47:3f` and identifier=1,32:76:8c:29:47:3f to the mac address specified in the error message

**Comment by JanYunkai** [comment]
Mac Os Catalina 10.15.3
➜  ~ minikube start --memory=8192 --cpus=4 --vm-driver=hyperkit
😄  minikube v1.7.3 on Darwin 10.15.3
✨  Using the hyperkit driver based on user configuration
🔥  Creating hyperkit VM (CPUs=4, Memory=8192MB, Disk=20000MB) ...

💣  Unable to start VM. Please investigate and run 'minikube delete' if possible: creating host: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary error: could not find an IP address for ba:5d:59:26:f9:29

😿  minikube is exiting due to an error. If the above message is not useful, open an issue:
👉  https://github.com/kubernetes/minikube/issues/new/choose

**Comment by dallas-lbg** [comment]
I'm seeing the same error as @JanYunkai

**Comment by fezho** [comment]
Got same issue with `minikube v1.7.3 on Darwin 10.14.6`, fix this by adding argument `--vm-driver=virtualbox`

**Comment by reeseann** [comment]
> Seeing same issue with a fresh install of docker for mac and minikube. It seems to be consistent (I tried uninstalling and reinstalling at least 5 times and got the same error each time). Also using mohave and do have a company VPN software installed if that is the cause (f5 Big-IP edge client).

Same issue, same configuration as @nickwilmes except Catalina. As a workaround, the virtualbox driver seems to be working (thx @fezho)

**Comment by Tony-Hu** [comment]
I also met the issue when I connected with my Cisco AnyConnect VPN. Resolved by disconnect the VPN.

**Comment by winhung** [comment]
Apparently this was caused by the McAfee Total Protection that i have installed on my macbook (Catalina v10.15.3).Disabling the Firewall function, resolved this issue.

**Comment by rodrigc** [comment]
@tstromberg I encountered the same problem on macOS Catalina 10.15.4 :

```
Temporary Error: open /var/db/dhcpd_leases: no such file or directory
```

On my machine, I had installed the following from homebrew:

* `brew install docker`
* `brew install docker-machine`
* `brew install docker-machine-driver-hyperkit`
* `brew install hyperkit`

to work around the problem, I uninstalled everything:

* `brew uninstall docker`
* `brew uninstall docker-machine`
* `brew uninstall docker-machine-driver-hyperkit`
* `brew uninstall hyperkit`

Then I Installed Docker Desktop for Mac from here:
https://hub.docker.com/editions/community/docker-ce-desktop-mac/

After doing that, I had:

```
docker version
Client: Docker Engine - Community
 Version:           19.03.8
 API version:       1.40
 Go version:        go1.12.17
 Git commit:        afacb8b
 Built:             Wed Mar 11 01:21:11 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.8
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.17
  Git commit:       afacb8b
  Built:            Wed Mar 11 01:29:16 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.13
  GitCommit:        7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```

I did **not** install docker-machine or docker-machine-driver-hyperkit.
It looks like Docker Desktop for Mac includes hyperkit.

At this point I was able to start docker containers and did not have the issue with the missing `/var/db/dhcpd_leases` file.

**Comment by jayeshd7** [comment]
when I am following below steps I got success -

1.Disconnect any VPN
2.minikube delete && minikube start --vm-driver=hyperkit
3. minikube start
4. (base) 90-X50XLVDQ-3GN:PycharmProjects jayeshdilip.dalal$ minikube start
😄  minikube v1.9.2 on Darwin 10.15.3
✨  Using the hyperkit driver based on existing profile
👍  Starting control plane node m01 in cluster minikube
🏃  Updating the running hyperkit "minikube" VM ...
🐳  Preparing Kubernetes v1.18.0 on Docker 19.03.8 ...
🌟  Enabling addons: default-storageclass, storage-provisioner
🏄  Done! kubectl is now configured to use "minikube"

❗  /usr/local/bin/kubectl is v1.15.5, which may be incompatible with Kubernetes v1.18.0.
💡  You can also use 'minikube kubectl -- get pods' to invoke a matching version

even check in 
cat /var/db/dhcpd_leases

there is IP address.

**Comment by priyawadhwa** [comment]
Is anybody still seeing this issue with the latest release (v1.9.2)?

**Comment by colangelo** [comment]
Yes, me, and I guess many of the others that commented here.
Nothing changed in the latest releases.
The only way that I have to make it work is to reboot, and then start minikube before starting the VPN.
Once I start the VPN, there is no way to have minikube start until I reboot: disconnecting the VPN does not fix the problem.

**Comment by tstromberg** [comment]
If you get this error, please try `--driver=docker` as a workaround: particularly if you are a VPN user! 

In the end, there is not a whole lot more we can do about this particular error, as it is internal to hyperkit. Two observations I've had are:

* This error can be related to Internet Sharing (hyperkit uses the DHCP server from it)
* This error can happen if two VM's are started simultaneously. We've now prevented this in the minikube code base.

I'm going to leave this open as a long-term support issue, as users are likely to hit it for years to come.

**Comment by MansM** [comment]
@winhung have your mcafee team create a rule in the firewall:

Rule Action: | Allow
-- | --
Direction: | Either
Connection type: | All types (Wired,Wireless,Virtual)
Protocol: | UDP/IPv4
Local port: | 67-68
Remote port: | 67-68

**Comment by niketjsh** [comment]
I tried several times this issue exists when you first create ingress service on the minikube
Its not particular to any of the minikube driver setup.

I know my solution would not be a best choice but i worked a several times using different virtual machines like vritualbox, hyperkit and  Docker Kubernetes

Here's my solution
1) Delete the minikube from any of the vertual machine you are running
2) Install helm using exact same instruction given in the https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-helm/
3)  Start the minikube using any vertual machine ( minikube start --vm-driver='driver')
4) apply your nginx ingress service using kubectl.

I believe this solution would work for sure.

**Comment by colangelo** [comment]
Looks like for me it's solved.
Running on macOS 10.15.6 (19G2021) and minikube v1.12.3 and I'm not experiencing the problem anymore, while before on my machines it was failing almost 100% of the times.

**Comment by Downchuck** [comment]
On Catalina (10.15.6) I'm not seeing a `dhcpd_leases` file written, and so the loop never completes -- I can of course write that file, to bump the loop forward, though it'll never get to a reachable, routed IP address. Firewall tested on and off, internet sharing tested on and off.

Docker hyperkit did need a bump, homebrew didn't seem to have the right version:
```
sudo curl https://github.com/machine-drivers/docker-machine-driver-hyperkit/releases/download/v1.0.0/docker-machine-driver-hyperkit -Lo /usr/local/bin/docker-machine-driver-hyperkit
```

Definitely needed to bump hyperkit (so maybe I made a mistake in my homebrew download):
```
brew upgrade hyperkit
```

Seems like there's a long standing issue somewhere in here about setting up networking.

**Comment by rodrigc** [comment]
@Downchuck the other alternative is to completely uninstall Homebrew packages of Docker and Hyperkit and instead install Docker Desktop for Mac:

https://github.com/kubernetes/minikube/issues/4206#issuecomment-605590333

**Comment by Downchuck** [comment]
@rodrigc there are many alternatives, such as using virtualbox, parallels or installing Docker for Mac. Using an alternative does not address the long standing issues, unless it is demonstrating a fix. If there is a fix in the Docker for Mac implementation it would be very helpful to understand it so as to address the breakdown here.

**Comment by hughpearse** [comment]
I encountered this issue after upgrading from catalina to big sur. Tried a bunch of stuff and eventually got it working again. Not sire exactly what I did, but I can show everything:

```bash
foo@bar$ brew reinstall hyperkit
foo@bar$ brew link hyperkit
foo@bar$ sudo chown root:wheel /usr/local/Cellar/hyperkit/0.20200908/bin/hyperkit
foo@bar$ sudo chmod u+s /usr/local/Cellar/hyperkit/0.20200908/bin/hyperkit
foo@bar$ brew reinstall docker-machine-driver-hyperkit
foo@bar$ sudo chown root:wheel /usr/local/opt/docker-machine-driver-hyperkit/bin/docker-machine-driver-hyperkit
foo@bar$ sudo chmod u+s /usr/local/opt/docker-machine-driver-hyperkit/bin/docker-machine-driver-hyperkit
foo@bar$ sudo rm /var/db/dhcpd_leases
foo@bar$ sudo touch /var/db/dhcpd_leases
foo@bar$ minishift stop
foo@bar$ minishift delete --force --clear-cache
foo@bar$ rm -rf ~/.minishift
foo@bar$ brew install --cask --force minishift
```

Go to System Preferences > Security & Privacy > Firewall > [Firewall Options] > [ ] Enable stealth mode

And disable stealth mode if enabled.

```bash
foo@bar$ minishift start --show-libmachine-logs -v5
```

I dont know for sure, but I suspect the fix was `brew install --cask --force minishift`

**Comment by javsalgar** [comment]
Same issue in Big Sur. I tried disconnecting from VPNs, tried the steps from @hughpearse but no luck

**Comment by Mars-2030** [comment]
I am still having teh same issue with BigSur .. it seems noting worked form me.. have any one fine a solution for that

**Comment by gilbahat** [comment]
not working for me as well.
as multipass are considering switching to the qemu driver, maybe minikube can support it too?
https://github.com/canonical/multipass/issues/2245

**Comment by sharifelgamal** [comment]
@gilbahat we are working on a qemu driver in the near future, specifically for arm64 support but just for more general support as well.

WRT to connectivity issues with hyperkit, it's something we're actively working on fixing in the short term: https://github.com/kubernetes/minikube/issues/3036

**Comment by amirsalaar** [comment]
for my case, I had my corporation VPN on (particularly McAfee), which was blocking the creation of `dhcpd_leases`.  After turning that off and shutting the firewall temporarily , it got fixed.

**Comment by edrandall** [comment]
Running MacOS Big Sur 11.6.2 in a corporate environment which has the `pf` firewall configured, have been experiencing this issue.

I created a new pf rules file `/etc/pf.anchors/io.k8s.signs.minikube` and whittled it down to the following rule:
```
# log is optional and uses `pflog0` by default;  `man pflog` gives instructions to see the logged packets
pass log on bridge100 inet proto udp from any port bootpc to port bootps
```

Reference the rules file in /etc/pf.conf: 
```
anchor "minikube"
load anchor "minikube" from "/etc/pf.anchors/io.k8s.sigs.minikube"
```

To reload pf using the updated rules:
```
# pfctl -d
# pfctl -e -f /etc/pf.conf
```

**Comment by georhiireva** [comment]
Hello, colleagues. 
MacOs BigSur.
I had the same issue with minishift and used recommendations from this topic to fix. I tried all options and this works for me (just use minikube instead of minishift):
1. minishift stop
2. minishit delete --clear-cahce
3. rm -fr ~/.minishift
4. brew reinstall hyperkit
5. brew reinstall docker-machine-driver-hyperkit
6. sudo chown root:wheel /usr/local/bin/docker-machine-driver-hyperkit &&
sudo chmod u+s /usr/local/bin/docker-machine-driver-hyperkit
7. sudo chown root:wheel /usr/local/bin/hyperkit &&
sudo chmod u+s /usr/local/bin/hyperkit
8. minishift config set vm-driver hyperkit (u'll have message: instance doesn't exist bla bla, will be set on next start - it's ok)
9. minishift start
Minishift start successfully.
PS: Check the location of docker-machine-driver-hyperkit and hyperkit before points 6 and 7.
I hope this will help somebody.

**Comment by cr-tech-repo** [comment]
@georhiireva - There is typo in your second command with the work minishift

**Comment by georhiireva** [comment]
> @georhiireva - There is typo in your second command with the work minishift and cache

@cr-tech-repo , thanks, fix it

**Comment by kanwarysingh** [comment]
> Running MacOS Big Sur 11.6.2 in a corporate environment which has the `pf` firewall configured, have been experiencing this issue.
> 
> I created a new pf rules file `/etc/pf.anchors/io.k8s.sigs.minikube` and whittled it down to the following rule:
> 
> ```
> # log is optional and uses `pflog0` by default;  `man pflog` gives instructions to see the logged packets
> pass log on bridge100 inet proto udp from any port bootpc to port bootps
> ```
> 
> Unsure whether the `bridge100` device is hard-coded or might change in future, would like to keep the rule as specific as possible.
> 
> Reference the rules file in /etc/pf.conf:
> 
> ```
> anchor "minikube"
> load anchor "minikube" from "/etc/pf.anchors/io.k8s.sigs.minikube"
> ```
> 
> To reload pf using the updated rules:
> 
> ```
> # pfctl -d
> # pfctl -e -f /etc/pf.conf
> ```
> 
> Cleared everything down with `minikube delete` then restarted `minikube start`, the following appears in the pflog0 tcpdump:
> 
> ```
>  00:00:04.420007 rule 2.minikube.0/0(match): pass in on bridge100: 0.0.0.0.68 > 255.255.255.255.67: BOOTP/DHCP, Request from ca:7e:32:41:3f:16, length 278
> ```
> 
> The minikube starts successfully.

@edrandall I get a syntax error while enabling the firewall.

```
/etc/pf.conf:36: syntax error
pfctl: Syntax error in config file: pf rules not loaded
pfctl: load anchors
```
<img width="561" alt="Screen Shot 2022-01-31 at 1 05 33 PM" src="https://user-images.githubusercontent.com/15788034/151872991-d451160c-5b87-4177-91be-ba43d2641542.png">

My rule file contains:
```
# log is optional and uses `pflog0` by default;  `man pflog` gives instructions to see the logged packets
pass log on bridge100 inet proto udp from any port bootpc to port bootps
```

**Comment by edrandall** [comment]
probably a mismatched quote somewhere, could be in pf.conf or any of your anchor files.

**Comment by jvinceller** [comment]
I've tried all the solutions with a Mac M1 with Monterey and the x86 (Rosetta) execution mode and I couldn't get _minishift_ or _minikube_ working. It's always something at the network adapter. My Mac seems to have the leases at var/db/dhcpclient/leases and not under _/var/db/dhcp_leases_ anyway.

**Comment by julienbonastre** [comment]
Facing this issue too on BigSur Mac - tried all tricks about to no avail

**Comment by edrandall** [comment]
Try the following packet filter rules:

```
#
# Allows minikube to run using hyperkit
# /etc/pf.anchors/io.k8s.sigs.minikube - load by adding 2 lines to /etc/pf.conf:
# ---
# anchor "io.k8s.sigs.minikube"
# load anchor "io.k8s.sigs.minikube" from "/etc/pf.anchors/io.k8s.sigs.minikube"
# ---
# Restart: pfctl -d ; pfctl -e -f /etc/pf.conf
# To watch logged packets, add "log" directive to rule(s) and run:
# ifconfig pflog0 create
# tcpdump -n -e -ttt -i pflog0
#

# Allow hyperkit DHCP packet in
pass on bridge100 inet proto udp from any port bootpc to port bootps

# Allow hyperkit to pull images
pass inet proto udp from 192.168.0.0/16 to any port 53
pass inet proto tcp from 192.168.0.0/16 to any
```

**Comment by NaoyukiTai** [comment]
This is biting me as well. I followed up most of suggestions including the /etc/pf.anchors. I cannot turn off the firewall as my company Mac is under control of Falcon. 

Mac is Intel, Monterey 12.4.  As of 2022-07-18, as far as I can tell, 
minikube - 1.26.0
hyperkit  - 0.20200908
docker-machine-driver-hyperkit - 1.0.0

[minikube_logs_1b004fc441daa6a7016058454b974cf8421ba12a_0.log](https://github.com/kubernetes/minikube/files/9135824/minikube_logs_1b004fc441daa6a7016058454b974cf8421ba12a_0.log)

[logs.txt](https://github.com/kubernetes/minikube/files/9135826/logs.txt)

**Comment by maffydavison** [comment]
I'm on Mac Monterrey and saw the note about the dhcp_leases file not being there (true) but /var/db/dhcpclient containing IP information.  I'm in a crunch to get through this K8s tutorial, so I thought I'd try something down and dirty and copy one of the files from the dhcpclient to /var/db/dhcpd_leases and see if that worked -- but it couldn't parse XML, so I'm stuck again.  I'll clean up my hack and pull the logs - but the issue remains that there is no such file as dhcpd_leases

😿  Failed to start hyperkit VM. Running "minikube delete" may fix it: creating host: create: Error creating machine: Error in driver during machine creation: IP address never found in dhcp leases file Temporary error: unable to parse line: <?xml version="1.0" encoding="UTF-8"?
[minikube_logs_d5f6b3eb7402185b5cbe7c8e6ab1ab332f57ca4f_0.log](https://github.com/kubernetes/minikube/files/9427981/minikube_logs_d5f6b3eb7402185b5cbe7c8e6ab1ab332f57ca4f_0.log)
[logs.txt](https://github.com/kubernetes/minikube/files/9427982/logs.txt)
>

**Comment by caerulescens** [comment]
I'm going to investigate starting `hyperkit` on `macos`; I'll report what I find.

**Comment by divysinghvi** [comment]
hey @colangelo, Hyperkit has since been deprecated in Minikube you can try shifting to vfkit,qemu or krun
Closing this issue since it applies to a deprecated driver. If anyone experiences the same behavior with one of the supported drivers (vfkit, qemu, or krun), please feel free to re-open a new issue.

**Comment by nirs** [comment]
Hyperkit will be removed soon: #21601 

/close
